### PR TITLE
chore: add default params preference [f-10]

### PR DESCRIPTION
**Pull request by frederickfogerty**
## Before

There was no ability to add default parameters to product images that were being passed through

## After

Default parameters can be set on product images. Also I restructured the product images code a bit to make the fall-back case clearer.

### Stacked PR Chain: f-10
| PR | Title | Status |  Merges Into  |
|:--:|:------|:-------|:-------------:|
|#159|test: when imgixBaseURL isn't set, integration is no-op||-|
|#161|👉 chore: add default params preference||#159|

### CODE REVIEW

**Comment by frederickfogerty** [comment]
@luqven for some reason annotate isn't working here?

**Comment by luqven** [comment]
> @luqven for some reason annotate isn't working here?

Weird.. will investigate.

File: cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js [hunk 1]
```diff
@@ -10,36 +10,87 @@ const currentSite = require('dw/system/Site').getCurrent();
...
+        if (firstImage) {
+          result = [
+            {
+              alt: firstImage.alt,
+              url:
+                imgixBaseURL +
+                firstImage.URL.toString() +
+                (isBaseURLSet && imgixDefaultParams
+                  ? `?${imgixDefaultParams}`
+                  : ""),
```
Comment (luqven) [inline cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js:36]: nice 👏🏼

**Review by luqven — APPROVED** [review]
lgtm

File: cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js (hunk 1 above)
Comment (luqven) [reply cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js:36]: Something strange is happening where the string interpolation just isn't taking place 🤔 
`https://zybp-003-sandbox.imgix.net/on/demandware.static/-/Sites-apparel-m-catalog/default/dw4ca75192/images/large/PG.52001DAN84Q.BLACKWL.PZ.jpg?${imgixDefaultParams}`

File: cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js [hunk 2]
```diff
@@ -10,36 +10,87 @@ const currentSite = require('dw/system/Site').getCurrent();
...
+        var firstImage = collections.first(images);
+        if (firstImage) {
+          result = [
+            {
+              alt: firstImage.alt,
+              url:
+                imgixBaseURL +
+                firstImage.URL.toString() +
+                (isBaseURLSet && imgixDefaultParams
+                  ? `?${imgixDefaultParams}`
```
Comment (luqven) [inline cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js:35]: Turns out SFCC doesn't like ES6 very much 
```suggestion
                  ? "?" + imgixDefaultParams
```

File: cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js [hunk 3]
```diff
@@ -10,36 +10,87 @@ const currentSite = require('dw/system/Site').getCurrent();
...
-        this[type] = result;
+      } else {
+        result = collections.map(images, function (image, index) {
+          return {
+            alt: image.alt,
+            url:
+              imgixBaseURL +
+              image.URL.toString() +
+              (isBaseURLSet && imgixDefaultParams
+                ? `?${imgixDefaultParams}`
```
Comment (luqven) [inline cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js:51]:
```suggestion
                  ? "?" + imgixDefaultParams
```

**Review by luqven — CHANGES_REQUESTED** [review]
We need to change these from ES6 template literals to something that SFCC plays nice with.

**Comment by luqven** [comment]
@frederickfogerty Added to commits to fix a small bug and oversight. Hope that's alright 😉 .

**Comment by frederickfogerty** [comment]
Thanks for updating this @luqven 🙌

**Pull request merged**